### DISCUSSION TITLE
Encrypted telemetry

### DISCUSSION THREAD

**Discussion by mykls** [ExpressLRS/ExpressLRS#1861] [Ideas]
Hi.  I understand the Crossfire has an option for encrypted telemetry.  Is this something that could be added to ExpressLRS?

**Comment by commanderguy3001** [comment]
we've already talked about "encryption" in elrs a bunch of times, mostly for the main RC link, but telemetry has the same problems.
simple answer: No, it will not be added.
hard answer: it doesn't make sense to add encryption to elrs, for multiple reasons:
- it would slow the link way down
- it wouldn't be effective, as it would be easily breakable
- unless you would add a ton of extra data which would cripple the range (and also would degrade latency even further
- it is just a hobby grade system that is not designed to be "secure" in any way
- it is not allowed to have encrypted transmission in HAM bands
TBS doesn't care about the first point, XF isn't meant to be specifically fast in any way, neither do they care about the second point. They didn't add extra data to make it more secure from all I can tell, which comes from not caring about the second point, or just not knowing. The fourth point doesn't apply to them, they are a commercial system. The fifth point they can work around, by being a closed source thing you have to buy, and they have their stuff certified.

**Reply by mykls** [reply]
Makes sense.  Thanks.
